staging/src/k8s.io/api/flowcontrol/v1alpha1/types.go

```diff
@@ -0,0 +1,296 @@
+
+	// `items` is a list of flow-schemas.
+	Items []FlowSchema `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// FlowSchemaSpec describes how the flow-schema's specification looks like.
+type FlowSchemaSpec struct {
+	// `priorityLevelConfiguration` should reference a PriorityLevelConfiguration in the cluster. If the reference cannot
+	// be resolved, the flow-schema will be ignored and marked as invalid in its status.
+	// Required.
+	PriorityLevelConfiguration PriorityLevelConfigurationReference `json:"priorityLevelConfiguration,omitempty" protobuf:"bytes,1,opt,name=priorityLevelConfiguration"`
+	// `matchingPrecedence` is used to choose among the FlowSchemas that match a given request. The chosen
+	// FlowSchema is among those with the numerically lowest (which we take to be logically highest)
+	// MatchingPrecedence.  Each MatchingPrecedence value must be non-negative.
+	// Note that if the precedence is not specified or zero, it will be set to 1000 as default.
+	MatchingPrecedence int32 `json:"matchingPrecedence" protobuf:"varint,2,opt,name=matchingPrecedence"`
+	// `distinguisherMethod` defines how to compute the flow distinguisher for requests that match this schema.
+	// `nil` specifies that the distinguisher is disabled and thus will always be the empty string.
+	DistinguisherMethod *FlowDistinguisherMethod `json:"distinguisherMethod,omitempty" protobuf:"bytes,3,opt,name=distinguisherMethod"`
+	// `rules` describes which requests will match this flow schema. The flow-schema applies if any rule matches.
+	Rules []PolicyRuleWithSubjects `json:"rules,omitempty" protobuf:"bytes,4,rep,name=rules"`
+}
+
+// FlowDistinguisherMethodType is the type of flow distinguisher method
+type FlowDistinguisherMethodType string
+
+// FlowDistinguisherMethod specifies the method of a flow distinguisher.
+type FlowDistinguisherMethod struct {
+	// `type` is the type of flow distinguisher method
+	Type FlowDistinguisherMethodType `json:"type,omitempty" protobuf:"bytes,1,opt,name=type"`
+}
+
+// PriorityLevelConfigurationReference contains information that points to the "request-priority" being used.
+type PriorityLevelConfigurationReference struct {
+	// `name` is the name of resource being referenced
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+}
+
+// PolicyRuleWithSubjects prescribes a test that applies to a request to an apiserver. The test considers the subject
+// making the request, the verb being requested, and the resource to be acted upon.
+type PolicyRuleWithSubjects struct {
+	// `subjects` is the list of normal user, serviceaccount, or group that this rule cares about.
+	// +optional
+	Subjects []Subject `json:"subjects,omitempty" protobuf:"bytes,1,opt,name=subjects"`
+	// `rule` is the target verb, resource or the subresource the rule cares about. APIGroups, Resources, etc.
+	// Required.
+	Rule PolicyRule `json:"rule" protobuf:"bytes,2,opt,name=rule"`
```

Comment:
in RBAC we chose to use a slice of these because you want to make sure the group and resource tuple expansions are correct. You don't want to match deployments.rbac.authorization.k8s.io for instance.

Reply:
That structure will require that subjects be defined multiple times to be able to grant access to resources in different groups.  That is overly onerous.